fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionSpec.java

```diff
@@ -64,7 +97,7 @@ public boolean isSatisfy(DistributionSpec spec) {
                 return false;
             }
         }
-        return hashDistributionDesc.isSatisfy(other.hashDistributionDesc);
+        return hashDistributionDesc.isSatisfy(other.hashDistributionDesc) || isJoinEqColumnsCompatible(other);
```

Comment:
I have question in here, check logical of `isStatisfy` will check `SHUFFLE_AGG`, `SHUFFLE_JOIN`, but you add the logical in here, it's looks will lose some check

Reply:
The original logical of `isSatisfy` is whether `item.columns.containsAll(this.columns)`. 

For `SHUFFLE_AGG` and `LOCAL`,  the logical is the same. And for other types, it required the two columns to be 
 exactly the same.